src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/ProvideSemanticTokensResponse.cs

```diff
@@ -29,12 +26,35 @@ public override bool Equals(object? obj)
             return false;
         }
 
-        return Tokens == other.Tokens ||
-              (Tokens is not null && other.Tokens is not null && Tokens.SequenceEqual(other.Tokens));
+        if (Tokens is not null && other.Tokens is not null)
+        {
+            if (Tokens?.Length != other.Tokens?.Length)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < Tokens!.Length; i++)
+            {
+                if (Tokens[i].Length != other.Tokens![i].Length)
+                {
+                    return false;
+                }
+
+                for (var j = 0; j < Tokens[i].Length; j++)
+                {
+                    if (Tokens[i][j] != other.Tokens[i][j])
+                    {
+                        return false;
+                    }
+                }
```

Comment:
Could this just be the following?

if (!Tokes[i].SequenceEqual(other.Tokens[i])
  return false;